### PR TITLE
add vochainPrivValidator flag to support a remote private validator

### PR DESCRIPTION
**Pull request by altergui**
with `tmkms` this enables the use of Ledger or Yubikey for privkey storage and signing

### CODE REVIEW

**Comment by altergui** [comment]
oops, i branched off `stage` instead of `master`, i'll rebase asap.
but feedback on the proposal already welcome.

**Comment by altergui** [comment]
here's the output, using `--vochainPrivValidator tcp://0.0.0.0:26658`
(while running at the same time `tmkms start -c kms/tmkms.toml -v`)
```
dvotenode_1  | 2021-10-29T11:21:30Z	INFO	vochain/start.go:264	PrivValidatorListenAddr defined, Tendermint will wait for a remote private validator connection
dvotenode_1  | 2021-10-29T11:21:30Z	INFO	vochain/start.go:297	tendermint p2p node ID: 145b2a2df939f4ccc306db0dfa4fb66360b9f7de
dvotenode_1  | 2021-10-29T11:21:30Z	INFO	vochain/start.go:301	found genesis file /app/run/stage/vochain/config/genesis.json
```

**Comment by p4u** [comment]
LGTM! @jordipainan?

**Review by jordipainan — APPROVED** [review]

**Comment by jordipainan** [comment]
> LGTM! @jordipainan?

LGTM too! Thank you @altergui

**Comment by jordipainan** [comment]
@altergui can we change the base to master ? We use it as dev branch :)

**Comment by altergui** [comment]
definitely, i spotted that last friday but thought there would be conflicts in a rebase and would take me some time... in fact it was trivial. here you go

**Comment by altergui** [comment]
a minor 'regression' in log output is that when this flag is enabled, the log doesn't print the pubkey anymore (since it's not yet known).
i could make it print the pubkey after node initialization, since i guess it's convenient for debugging.

**Comment by jordipainan** [comment]
perf! yup would be nice to print the pubkey, I do not think is a big deal to print it after initialization :)

File: vochain/start.go [hunk 1]
```diff
@@ -361,3 +379,9 @@ func AminoKeys(key crypto25519.PrivKey) (private, public []byte, err error) {
 
 	return aminoKey, aminoPubKey, nil
 }
+
+// AminoPubKey is a helper function that transforms a standard EDDSA pubkey into
+// Tendermint like amino format useful for creating genesis files.
+func AminoPubKey(pubkey []byte) (public []byte, err error) {
```
Comment (jordipainan) [inline vochain/start.go]: As public and err are not assigned here you can just have `func AminoPubKey(pubKey []byte) ([]byte, error)`

**Comment by p4u** [comment]
What's missing here @altergui @jordipainan ? can we merge it?

File: vochain/start.go (hunk 1 above)
Comment (altergui) [reply vochain/start.go]: nice, thanks! amended

**Comment by altergui** [comment]
i think it's safe enough to merge, shouldn't introduce new bugs unless the flag is used :)

**Pull request merged**